### DISCUSSION TITLE
How do I use sendEmail command instead of sendmail or enabling MTA on system

### DISCUSSION THREAD

**Discussion by Blason** [fail2ban/fail2ban#3389] [Q&A]
Hi team,

Can I enable customized emails instead of default templates with my own custom jails? I mean there is a command called sendEmail which does send a email without any need to install MTA on server. can someone please confirm? And can I use global such command? 

below are my custom jails

[nginx-ips]
enabled = true
port = http,https
logpath = /var/log/nginx/isn/access.log
bantime = -1
findtime = 10m
maxretry = 1
backend = auto

**Comment by Blason** [comment]
Hi Team,

Can anyone suggest an answer me please?

**Reply by sebres** [reply]
I don't know what you mean as "sendEmail command", but...
If it is some shell command, fail2ban would need a [command-action](https://github.com/fail2ban/fail2ban/blob/master/config/action.d/) to be able to execute it, for an example see some other mail action, like [mail-whois-lines](https://github.com/fail2ban/fail2ban/blob/master/config/action.d/mail-whois-lines.conf)

If it is compatible to `mail` command, you can simply configure it with overwriting internal parameter of action `mailcmd`, soomething like this:
```ini
[nginx-ips]
...
action = %(action_)s
         mail-whois-lines[mailcmd="sendMail -param", sender="%(sender)s", dest="%(destemail)s", protocol="%(protocol)s", chain="%(chain)s"]
```
If it isn't compatible (or very different), rewrite some action so it will execute commands you need. For more details see the available actions or section ACTION CONFIGURATION FILES in [man jail.conf.5](https://manpages.debian.org/unstable/fail2ban/jail.conf.5.en.html#ACTION_CONFIGURATION_FILES_(action.d/*.conf)).

The jail would look similar (here for example if you call new action `action.d/sendEmail.conf`):
```ini
[nginx-ips]
...
action = %(action_)s
         sendEmail[sender="%(sender)s", dest="%(destemail)s", ..., other_param_the_action_may_ned="...", ...]
```


If it is something complex, you could write pythonic action instead of command action.

**Reply by Blason** [reply]
Hi,

And thanks for the reply. The sendEmail is a simple linux command by which you do not have to install a dedicated MTA on server but a client based utility. The only thing is; it takes lot many parameters for sending out a mail like this. Hence was wondering how do I use the same with my jail.

/usr/bin/sendEmail -f $FROMEMAIL -t $TOEMAIL -u $SUB -o tls=yes -s smtp.gmail.com:587 -xu $FROMEMAIL -xp $EPASS
